File: javascript/word-count/word-count.js
var Words = function() {};

Words.prototype.count = function(str) {
    var words = str.toLowerCase().split(/[ ,\n\t]+/);
    var result = {};

    for(i in words) {
        var current = words[i].trim();
        if (current) {
            result[current] = (isNaN(result[current]) ? 1 :  result[current] + 1);
        }
    }

    return result;
};

module.exports = Words;

File: java/word-count/src/main/java/WordCount.java
import java.util.HashMap;
import java.util.Map;

public class WordCount {

	public Map<String, Integer> phrase(String string) {
		String[] words = explode(string);
		Map<String, Integer> result = new HashMap<>();

		for(String word : words) {
			Integer count = result.putIfAbsent(word, 1);
			if (count != null)
				result.put(word, count+1);
		}
		
		return result;
	}

	private String[] explode(String string) {
		return string.toLowerCase().replaceAll("[^ a-z0-9]", " ").replaceAll("[ ]+", " ").split(" ");
	}

}